integration_tests/test_gravity.py

```diff
@@ -441,51 +452,35 @@ def test_gravity_cancel_transfer(gravity):
         amount = 1000
 
         print("send to cronos crc21")
-        recipient = HexBytes(ADDRS["community"])
-        send_to_cosmos(gravity.contract, erc20, recipient, amount, KEYS["validator"])
+        community = HexBytes(ADDRS["community"])
+        send_to_cosmos(gravity.contract, erc20, community, amount, KEYS["validator"])
         assert erc20.caller.balanceOf(ADDRS["validator"]) == balance - amount
 
         denom = f"gravity{erc20.address}"
-
         crc21_contract = None
 
-        def check_auto_deployment():
-            "check crc21 contract auto deployed, and the crc21 balance"
+        def local_check_auto_deployment():
             nonlocal crc21_contract
-            try:
-                rsp = cli.query_contract_by_denom(denom)
-            except AssertionError:
-                # not deployed yet
-                return False
-            assert len(rsp["auto_contract"]) > 0
-            crc21_contract = cronos_w3.eth.contract(
-                address=rsp["auto_contract"], abi=cronos_crc21_abi()
+            crc21_contract = check_auto_deployment(
+                cli, denom, cronos_w3, community, amount
             )
-            return crc21_contract.caller.balanceOf(recipient) == amount
-
-        def get_id_from_receipt(receipt):
-            "check the id after sendToChain call"
-            for _, log in enumerate(receipt.logs):
-                if log.topics[0] == HexBytes(
-                    abi.event_signature_to_log_topic(
-                        "__CronosSendToChainResponse(uint256)"
-                    )
-                ):
-                    return log.data
-            return "0x0000000000000000000000000000000000000000000000000000000000000000"
-
-        wait_for_fn("send-to-crc20", check_auto_deployment)
-
-        def check_fund():
-            v = crc21_contract.caller.balanceOf(ADDRS["community"])
-            return v == amount
+            return crc21_contract
+
+        wait_for_fn("send-to-crc21", local_check_auto_deployment)
 
-        wait_for_fn("send-to-ethereum", check_fund)
```

Comment:
this wait call is removed, but it seems not replaced with a new one?

Reply:
[check_auto_deployment](https://github.com/crypto-org-chain/cronos/blob/db79ab90c7a2108c26052f35440095a2e7cffbb4/integration_tests/test_gravity.py#L464) already check balance for [recipient](https://github.com/crypto-org-chain/cronos/blob/db79ab90c7a2108c26052f35440095a2e7cffbb4/integration_tests/test_gravity.py#L444) which is [community](https://github.com/crypto-org-chain/cronos/blob/db79ab90c7a2108c26052f35440095a2e7cffbb4/integration_tests/test_gravity.py#L480)